### PR TITLE
add knative-metrics-robot to coverage tool & config

### LINKED ISSUE #863: remove previous posts of knative-metrics-robot
remove previous posts of knative-metrics-robot

### PR DESCRIPTION
**Pull request by steuhs**
The rename from covbot to knative-metrics-robot prevents garbage collection of bot comments.

**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #863 

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: ci/prow/templates/prow_presubmit_gocoverate_job.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@
         - "--artifacts=$(ARTIFACTS)"
         - "--cov-threshold-percentage=[[.Base.GoCoverageThreshold]]"
         - "--github-token=/etc/covbot-token/token"
+        - "--posting-robot=knative-metrics-robot"
```
Comment (srinivashegde86) [inline ci/prow/templates/prow_presubmit_gocoverate_job.yaml]: do we need this, if that is the default?

Comment (steuhs) [reply ci/prow/templates/prow_presubmit_gocoverate_job.yaml]: In the future if bot name changes, only the template needs to be updated accordingly.

File: ci/prow/config.yaml [hunk 2]
```diff
@@ -246,6 +246,7 @@ presubmits:
         - "--artifacts=$(ARTIFACTS)"
         - "--cov-threshold-percentage=80"
         - "--github-token=/etc/covbot-token/token"
+        - "--posting-robot=knative-metrics-robot"
```
Comment (srinivashegde86) [inline ci/prow/config.yaml]: why are we adding this to the jobs, if its the default value? We should only be adding it if it deviates from default

File: ci/prow/templates/prow_presubmit_gocoverate_job.yaml (hunk 1 above)
Comment (srinivashegde86) [reply ci/prow/templates/prow_presubmit_gocoverate_job.yaml]: I understand the reqmt here. I am suggesting that we dont need to add this flag as that is already the default value in code

**Pull request merged**